### PR TITLE
Update to handle issues with constant restarts of dock

### PR DESCRIPTION
**Pull request by maverickames**
Changes
- Added feature to handle adding to dock with the position. Before it was just being added then would have to be moved.
- Added checks to better identify when items where not in the dock.
- Changed behavior to make all changes and restart dock at the end. The issue I was having is was inconsistencies because dock was not fully loaded before it was getting another kill command. This also reduces the number of restarts and flashing of the screen.

### CODE REVIEW

**Comment by maverickames** [comment]
Updated setup the linter on my system. The integration check is failing because I added the sudo permissions. @geerlingguy  is there a better way to do this?

File: roles/dock/tasks/dock-kill.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+- name: "Restart Dock."
+  ansible.builtin.command:
+    cmd: killall Dock
+  become: true
+  become_method: sudo
```
Comment (geerlingguy) [inline roles/dock/tasks/dock-kill.yml:6]: I don't think I'd like to specify the method here (to match style on other commands).

One thing I don't necessarily like about this approach is it requires `sudo` to work—whereas I don't believe it did before?

Comment (maverickames) [reply roles/dock/tasks/dock-kill.yml:6]: Yea its not my first choice. I do wonder how you have consist results with the current setup. If I run it without either the --no-reboot option or a several second wait it doesn't actually work for me. Many of the command show as ran but nothing changes. As soon as I put a wait or don't restart the dock on every change it works like a dream. 

Do you have suggestions or ideas why it works on your system?

Comment (maverickames) [reply roles/dock/tasks/dock-kill.yml:6]: @geerlingguy any thoughts on this?

Comment (ShadowHawk07) [reply roles/dock/tasks/dock-kill.yml:6]: @maverickames this PR fixed this role for me, now I can add 20+ dock items without any issues whereas before I could only reliably add 1 or 2. Thanks.

**Comment by geerlingguy** [comment]
It looks like the PR is failing on the restart Dock task, since it now restarts no matter what (seemingly)—it should not do it if not needed (e.g. if no dock items changed): 

```
TASK [geerlingguy.mac.dock : Restart Dock.] ************************************
changed: [localhost]
```

See: https://github.com/geerlingguy/ansible-collection-mac/runs/4161203102?check_suite_focus=true#step:9:204

**Comment by maverickames** [comment]
@geerlingguy Thanks for the response and the time you put into these projects. I wasn't sure if I should move forward with this solution or if you wanted to try something different. I am curious if you are getting consistent results with your setup? If you are happy with my solution and want me to move forward I will work on getting the restart fixed and submitted. 

@ShadowHawk07 thanks for confirming.

**Comment by aioue** [comment]
@maverickames Restart dock could be in a notify rather than always at the end, right? I'm getting unreliable dock item adding and positioning so keen to see this PR testing green!

**Pull request closed without merging**